velox/core/PlanNode.h

```diff
@@ -98,6 +98,14 @@ class PlanNode {
   virtual const std::vector<std::shared_ptr<const PlanNode>>& sources()
       const = 0;
 
+  /// Returns true if this is a leaf plan node and corresponding operator
+  /// requires an ExchangeClient to retrieve data. TableScanNode is a leaf node
+  /// that doesn't require an ExchangeClient. But ExchangeNode is a leaf node
+  /// that requires an ExchangeClient.
+  virtual bool requiresExchangeClient() const {
```

Comment:
Nice API with helpful documentation. Would you also mention it in the PR description?